### PR TITLE
feature/list-of-bought-rewards

### PR DESCRIPTION
**Pull request by rniedziolka**
Hi,

This is a short cut show how this feature works:

https://www.loom.com/share/4a8f5658e3d64e528ed972e346e4b4f2

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # frozen_string_literal: true
 
 class Order < ApplicationRecord
+  serialize :reward_snapshot
+
   belongs_to :employee
   belongs_to :reward
+
+  def snapshot_price
```
Comment (ciejj) [inline app/models/order.rb]: I think this name might be misleading -  I would rather use something like `transaction_price`, or simply `price`

File: app/views/orders/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<h2>Orders</h2>
+  <br>
+    <div class="row hstack gap-3">
+      <% @orders.each do |order| %>
+        <div class="card" style="width: 18rem">
+          <div class="card-body">
+            <h5 class="card-title"><%= order.reward_snapshot.title %></h5>
+            <p class="card-text"><%= order.reward_snapshot.description %></p>
+            <h6 class="card-subtitle mb-2 text-muted">Price: <%= order.reward_snapshot.price %></h6>
+            <h6 class="card-subtitle mb-2 text-muted">Purchase time: <%= order.created_at.strftime("%d-%m-%y %H:%M") %></h6>
```
Comment (ciejj) [inline app/views/orders/index.html.erb]: Maybe consider `using time_ag_in_words` helper for displaying time info https://apidock.com/rails/v6.0.0/ActionView/Helpers/DateHelper/time_ago_in_words

File: spec/system/admin_order_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+RSpec.describe 'AdminReward crud', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let!(:admin_user) { create(:admin_user) }
+  let!(:employee) { create(:employee) }
+  let!(:reward) { create(:reward, price: 1) }
+  let(:reward2) { create(:reward, price: 2) }
```
Comment (ciejj) [inline spec/system/admin_order_spec.rb:13]: do we really need to create the whole object, if you are using only it's price as an input value placeholder?
If you want to keep that value in the let for the sake of visibility consider something like:
```suggestion
  let(:new_price) { 2 }
```

File: spec/system/order_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Reward', type: :system do
```
Comment (ciejj) [inline spec/system/order_spec.rb]: I think it was supposed to be something like this:
```suggestion
RSpec.describe 'Order', type: :system do
```

**Review by ciejj — APPROVED** [review]
Well done - 👍  for the feature presentation in the video - it makes CR easier :)

**Comment by rniedziolka** [comment]
Hi,

Some fixes:
https://github.com/rniedziolka/employee-recognition-platform-1637255604/pull/21/commits/b6ab78a942bd6feaaee05b8678e04d71516eed43
@ciejj what is "CR"? :)

File: app/views/orders/index.html.erb (hunk 2 above)
Comment (oskarlakner) [reply app/views/orders/index.html.erb]: I like to use I18n.l to format time. You could google `localize time format rails`

**Review by oskarlakner — APPROVED** [review]

File: app/views/orders/index.html.erb (hunk 2 above)
Comment (rniedziolka) [reply app/views/orders/index.html.erb]: Implemented - thanks!

**Pull request merged**